File: java_membership_api/src/main/java/com/example/membership_api/controller/OrderController.java
package com.example.membership_api.controller;

import java.util.Map;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.PutMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

import com.example.membership_api.model.Order;
import com.example.membership_api.service.OrderService;
import com.example.membership_api.utils.JwtUtil;
import com.example.membership_api.utils.ResponseUtil;

import jakarta.servlet.http.HttpServletRequest;

@RestController
@RequestMapping("/orders")
public class OrderController {

    @Autowired
    private OrderService orderService;

    @GetMapping("/me")
    public ResponseEntity<Map<String,Object>> getMyOrders(
            @RequestParam(defaultValue="1") int page,
            @RequestParam(defaultValue="10") int perPage,
            HttpServletRequest request) {
        String userId = JwtUtil.extractUserIdFromRequest(request);
        var items = orderService.getAll(page, perPage, null, userId);
        var total = orderService.countAll(null, userId);
        return ResponseEntity.ok(Map.of("items", items, "total", total, "page", page, "perPage", perPage));
    }

    @GetMapping("")
    public ResponseEntity<Map<String,Object>> getAllOrders(
            @RequestParam(defaultValue="1") int page,
            @RequestParam(defaultValue="10") int perPage,
            @RequestParam(required=false) String status,
            @RequestParam(required=false, name="user_id") String userId) {
        var items = orderService.getAll(page, perPage, status, userId);
        var total = orderService.countAll(status, userId);
        return ResponseEntity.ok(Map.of("items", items, "total", total, "page", page, "perPage", perPage));
    }

    @GetMapping("/{id}")
    public ResponseEntity<Order> getOne(@PathVariable String id) {
        return orderService.getById(id).map(ResponseEntity::ok)
                .orElse(ResponseEntity.notFound().build());
    }

    @PostMapping("")
    public Object create(HttpServletRequest request,@RequestBody Map<String, Object>  data) {
        String userId = JwtUtil.extractUserIdFromRequest(request);
        data.put("user_id",userId);
        return ResponseEntity.ok(ResponseUtil.success("Order Created Succefully",orderService.create(data)));
    }

    @PutMapping("/{id}")
    public ResponseEntity<Order> update(@PathVariable String id, @RequestBody Order order) {
        return orderService.update(id, order)
                .map(ResponseEntity::ok)
                .orElse(ResponseEntity.notFound().build());
    }

    @PutMapping("/confirm/{id}")
    public ResponseEntity<?> confirm(@PathVariable String id, HttpServletRequest request) {

        String userId = JwtUtil.extractUserIdFromRequest(request);
        try{
            return orderService.confirm(userId, id)
                .map(ResponseEntity::ok)
                .orElse(ResponseEntity.notFound().build());
        } catch (RuntimeException ex) {
            return ResponseUtil.error(ex.getMessage(), 400);
        }
    }
}

File: flask_api/app/controllers/order_controller.py
from flask import request
from flask_jwt_extended import jwt_required, get_jwt_identity
from app.services.order_service import OrderService
from app.utils.response import success_response, error_response


class OrderController:

    @staticmethod
    @jwt_required()
    def get_my_orders():
        user_id = get_jwt_identity()
        page = request.args.get("page", default=1, type=int)
        per_page = request.args.get("per_page", default=10, type=int)

        filters = {"user_id": user_id}

        filters = {k: v for k, v in filters.items() if v is not None}

        orders, total = OrderService.get_all_for_user(page, per_page, filters)
        return success_response(
            "Orders retrieved",
            {
                "items": [o.to_dict() for o in orders],
                "total": total,
                "page": page,
                "per_page": per_page,
            },
        )

    @staticmethod
    @jwt_required()
    def get_all_orders():
        page = request.args.get("page", default=1, type=int)
        per_page = request.args.get("per_page", default=10, type=int)

        filters = {
            "status": request.args.get("status"),
            "user_id": request.args.get("request_id"),
        }

        filters = {k: v for k, v in filters.items() if v is not None}

        orders, total = OrderService.get_all_orders(page, per_page, filters)
        return success_response(
            "Orders retrieved",
            {
                "items": [o.to_dict() for o in orders],
                "total": total,
                "page": page,
                "per_page": per_page,
            },
        )

    @staticmethod
    @jwt_required()
    def get_one(order_id):
        order = OrderService.get_by_id(order_id)
        if not order:
            return error_response("Order not found", 404)
        return success_response("Order found", order.to_dict())

    @staticmethod
    @jwt_required()
    def create():
        data = request.get_json()
        data["user_id"] = get_jwt_identity()
        order = OrderService.create(data)
        return success_response("Order created", order.to_dict())

    @staticmethod
    @jwt_required()
    def update(order_id):
        data = request.get_json()
        updated = OrderService.update(order_id, data)
        if not updated:
            return error_response("Order not found", 404)
        return success_response("Order updated", updated.to_dict())

    @staticmethod
    @jwt_required()
    def confirm_order(order_id):
        user_id = get_jwt_identity()
        updated = OrderService.confirm(user_id, order_id)
        if not updated:
            return error_response("Order not found", 404)
        return success_response("Order updated", updated.to_dict())
